### PR TITLE
review-branch

### PR DESCRIPTION
**Pull request by ashmalzahra**
To-Do List functionalities:

- Add task
- Remove task
- Clear all completed tasks
- Edit tasks
- Change the value of completed in the DOM when user checks the checkbox

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  e.preventDefault();
+  const newItem = document.getElementById('data');
+  if (newItem.value) {
+    const description = newItem.value.trim();
+    const index = todolist.length + 1;
+    newTodoList.addTask(description, index);
+    localStorage.setItem('todolist', JSON.stringify(newTodoList));
+    newItem.value = '';
+  }
+  window.location.reload();
```
Comment (anderCM) [inline index.js:39]: Please Kindly avoid using `window` to reach the `location` method, You can use `location` directly like this: `location.reload();`

File: index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  toggleStatus(task);
+  newTodoList.todolist[i - 1] = task;
+  localStorage.setItem('todolist', JSON.stringify(newTodoList));
+};
+
+const tasks = document.querySelectorAll('.task');
+tasks.forEach((e) => {
+  const checkInput = e.childNodes[0];
+  checkInput.addEventListener('change', checkBox);
+});
```
Comment (anderCM) [inline index.js:91]: According to `KISS` rules, get all the `task` classes from the `DOM` just to select one of them is not a good practice. So please kindly add an `id`, a `name` or another way to identify this element get only this element to apply the `eventListener`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+const clearCompleted = () => {
+  const filteredList = newTodoList.todolist.filter((e) => e.completed === false);
+  const sortedList = filteredList.map((object, i) => {
+    const index = i + 1;
+    return { ...object, index };
+  });
+  newTodoList.todolist = sortedList;
+  localStorage.setItem('todolist', JSON.stringify(newTodoList));
+  window.location.reload();
```
Comment (anderCM) [inline index.js:103]: Please Kindly avoid using `window` to reach the `location` method, You can use `location` directly like this: `location.reload();`

**Review by anderCM — CHANGES_REQUESTED** [review]
# Highlights!
- Good feature branch name ✔️
- Descriptive pull request ✔️
- Descriptive commit messages ✔️
- Good code organization and working app ✔️
# Status: Required Changes ♻️⛔️
Check the comments under the review.

**Comment by ashmalzahra** [comment]
Requested changes implemented. kindly review project!

**Review by anderCM — COMMENTED** [review]
# Highlights!
Good feature branch name ✔️
Descriptive pull request ✔️
Descriptive commit messages ✔️
Good code organization and working app ✔️

# Status: Approbed ♻️🎉🎉

Hi @ashmalzahra , your project is complete. There is nothing else to say than merge it!

**Comment by ashmalzahra** [comment]
Thank you! :memo:

**Pull request merged**